pkg/k8sresource/funcs.go

```diff
@@ -1,13 +1,40 @@
 package k8sresource
 
-import "k8s.io/apimachinery/pkg/types"
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
 
-func Kind(o Object) string {
-	return o.GetObjectKind().GroupVersionKind().Kind
+func DeleteAnnotation(o Object, key string) {
+	a := o.GetAnnotations()
+	if a == nil {
+		return
+	}
+
+	delete(a, key)
+}
```

Comment:
You still need to set the annotations, so
```suggestion
	o.SetAnnotations(a)
}
```

Reply:
It was working because the map is actually a pointer to something. But that was a lucky conincidence.